cluster-autoscaler/cloudprovider/azure/azure_cache.go

```diff
@@ -117,10 +120,22 @@ func (m *azureCache) regenerate() error {
 		}
 	}
 
+	// Regenerate VMSS to autoscaling options mapping.
+	newAutoscalingOptions := make(map[azureRef]map[string]string)
+	for _, vmss := range m.scaleSets {
+		ref := azureRef{Name: *vmss.Name}
+		options := extractAutoscalingOptionsFromScaleSetTags(vmss.Tags)
+		if !reflect.DeepEqual(m.getAutoscalingOptions(ref), options) {
+			klog.V(4).Infof("Extracted autoscaling options from %q ScaleSet tags: %v", *vmss.Name, options)
+		}
+		newAutoscalingOptions[ref] = options
```

Comment:
nit: should this be done in the above if check?

Reply:
We're rebuilding that option cache from scratch (`newAutoscalingOptions` is empty before that loop) so we have to store both changed and previously known options there.

Reply:
oh gotcha